### ISSUE TITLE
UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown

### ISSUE THREAD

**Issue by manwarA** [posit-dev/positron#10527]
## System details:
Win 11 Business, version 25H2
Experience: Windows Feature Experience Pack 1000.26100.253.0
#### Positron and OS details:

Positron Version: 2025.11.0 (user setup) build 234
Code - OSS Version: 1.105.0
Commit: https://github.com/posit-dev/positron/commit/807acdfca108da6b9b2d96452fe047238da1bd9f
Date: 2025-10-31T23:21:13.588Z
Electron: 37.6.0
Chromium: 138.0.7204.251
Node.js: 22.19.0
V8: 13.8.258.32-electron.0
OS: Windows_NT x64 10.0.26200

#### Session details:

Python 3.12.9 (System) started.
Python 3.12.9 (tags/v3.12.9:fdb8142, Feb  4 2025, 15:27:58) [MSC v.1942 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.18.1 -- An enhanced Interactive Python. Type '?' for help.

## Describe the issue:

When I create plot, some plots appear in the plot panel, while for others, I am having this issue:

`
"FigureCanvasAgg is non-interactive, and thus cannot be shown".
`

Its been a while that I am trying to figure out what's wrong with the setting or way I am plotting, however, I could not find reasonable answer from google, SO or reddit. Even the simplest thing like very basic plot is showing the error

`
plt.plot([1,2,3,4,5])
matplotlib.lines.Line2D at 0x2213533c6b0

plt.show()
positron-console-cell-217>:1: UserWarning:
FigureCanvasAgg is non-interactive, and thus cannot be shown
`

After several failed attempts, I am seeking help so that I can rectify this issue. I am sorry, if it has been solved, but I could not find anything. My positron is latest.

Plus, multiple plots are stacked in one plot (being displayed as one), even though if I closed the tab, the new plot will carry all the previous plots and the error. 

## Steps to reproduce the issue:

After 4 or 5 figures in the PLOTS panel, it starts giving me this error, and it is not consistent either. Now I have lost my figure panel altogether. 
For example, this happened now: 

>>> fpr, tpr, threshold = skm.roc_curve(y_1, predicted)
>>> roc_auc = skm.auc(fpr, tpr)
>>> plt.title('Receiver Operating Characteristic Curve')
Text(0.5, 1.0, 'Receiver Operating Characteristic Curve')
>>> plt.plot(fpr, tpr, 'b', label = 'AUC = %0.2f' % roc_auc)
[<matplotlib.lines.Line2D at 0x23694d0ac00>]
>>> plt.legend(loc = 'lower right')
<matplotlib.legend.Legend at 0x2363834c3e0>
>>> plt.plot([0, 1], [0, 1],'r--')
[<matplotlib.lines.Line2D at 0x236311fbe60>]
>>> plt.xlim([0, 1])
(0.0, 1.0)
>>> plt.ylim([0, 1])
(0.0, 1.0)
>>> plt.ylabel('True Positive Rate')
Text(9.444444444444438, 0.5, 'True Positive Rate')
>>> plt.xlabel('False Positive Rate')
Text(0.5, 9.444444444444459, 'False Positive Rate')
>>> ax = plt.gca()
>>> clf_disp = RocCurveDisplay.from_estimator(logr, x_1, y_1)
>>> clf_disp.plot(ax = ax)
<sklearn.metrics._plot.roc_curve.RocCurveDisplay at 0x23631214290>
>>> plt.plot([1,2,3,4,5])
[<matplotlib.lines.Line2D at 0x23694324ad0>]
>>> plt.show()
<positron-console-cell-76>:1: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown

## Expected or desired behavior:
1) No error
2) Figure display without plt.show()
3) Each figure in new canvas or panel. 

@juliasilge

**Comment by midleman** [comment]
Thanks for the detailed report! I was able to reproduce part of your issue - specifically the plots overlaying/stacking on top of each other (which is tracked in #7591). When creating multiple plots without explicitly creating new figures, they overlay on the same canvas.

Regarding the plt.show() warning:
I didn't encounter the "FigureCanvasAgg is non-interactive" warning in my testing. This might be environment-specific (Windows vs macOS) or related to your matplotlib  version/configuration.

Solution for both issues:
The good news is there's a single workaround that addresses both problems:
  1. Don't use plt.show() in Positron - plots automatically display in the Plots panel
  2. Use plt.figure() before each new plot to create separate visualizations

  Example:
```
  import matplotlib.pyplot as plt

  # Plot 1
  plt.figure()  # Create new figure
  plt.plot([1, 2, 3, 4, 5])
  plt.title('Plot 1')
  # NO plt.show() needed

  # Plot 2
  plt.figure()  # Create NEW figure
  plt.plot([5, 4, 3, 2, 1])
  plt.title('Plot 2')
  # NO plt.show() needed
```
Could you try this and let us know if it helps? Also, if you could share your matplotlib version (`import matplotlib; print(matplotlib.__version__)`), that would help us understand the warning behavior.

**Comment by manwarA** [comment]
The matplot version is:

```{python}
matplotlib.__version__
'3.10.0'
```

The above solution did not fix the problem. 

The following code works great, no need to display the figure (different lib though). 
```{python}
import plotly.express as px

sc = StandardScaler()
arr_fit = sc.fit_transform(gse60331_prep.transpose())

# apply PCA 
pca = PCA(n_components=3)
arr_fit_pc = pca.fit_transform(arr_fit)

explained_variance = pca.explained_variance_ratio_

labels = {
    str(i): f"PC {i+1} ({var:.1f}%)"
    for i, var in enumerate(pca.explained_variance_ratio_ * 100)
}

fig = px.scatter_matrix(
    arr_fit_pc,
    labels=labels,
    dimensions=range(3),
    color=gse60331_outcome["response"],
    color_discrete_map= {"Responder" : "#D2691E", "Non-responder" : "#5F9EA0"}
)
fig.update_traces(diagonal_visible=False)
# fig.show(), no need for this, automatically display the figure
```

However, this does not work et al. 
```{python}
>>> plt.figure()
<Figure size 640x480 with 0 Axes>
>>> plt.plot([1,2,3,4,5]) # does not draw the plot, so call the plt.show()
[<matplotlib.lines.Line2D at 0x1df9d85d370>]
>>> plt.show() # here it is, same error
<positron-console-cell-110>:1: UserWarning:

FigureCanvasAgg is non-interactive, and thus cannot be shown

```

Plt based-things does not work. Any other suggestion or any further information you need. 

Here, how I call the libraries (if that matters):
```{python}
import matplotlib.pyplot as plt
import matplotlib as mpl

mpl.rcParams['agg.path.chunksize'] = 100000
```

**Comment by juliasilge** [comment]
Can you share in more detail what you are seeing? I am not able to reproduce the problems you are describing. For example, if I do this:

```python
import matplotlib.pyplot as plt
plt.plot([1,2,3,4,5])
```

Then this is what I see:

<img width="1495" height="980" alt="Image" src="https://github.com/user-attachments/assets/0ce0f797-63bd-4029-a342-669ba2ddbdce" />

Do you see a problem with trying to run this very simple code? Can you share a screenshot or video of what you are observing?

**Comment by manwarA** [comment]
Hi, I have restarted the positron few times, the issue did not go away. However, when I open the new window, the code worked as expected (Figure is attached, code is working in new window). 

<img width="2514" height="1381" alt="Image" src="https://github.com/user-attachments/assets/f156ae91-1a14-42c3-8b5d-4776f18eff5f" />


I am attaching a movie just to let you know what I am facing. In this video, you can see that the code does not work. I might have messed up something in the settings of positron or something else, I really don't know (even restarting positron does not work). 

(https://drive.google.com/file/d/1H2A8WLEbjS8M1ySUNo5whdGaUgvHG88t/view?usp=drive_link)

**Comment by juliasilge** [comment]
@manwarA I feel a bit confused; are you saying that the simple code example does work, or does not work? Or maybe works for you intermittently?

Can you attach the video to the GitHub issue itself? Or make that Google Drive link accessible to the public, so we can see it?

**Comment by manwarA** [comment]
Sorry, I thought the link is accessible. Please try this one. 

https://drive.google.com/file/d/1tgeIhpj_2GkUL7UXAIe1rBrvIqnK2j8b/view?usp=sharing

You can see if I open the new positron window, the code works without problem. However, with the positron window where I am working, it does not work. 

I can open all the files in new positron window, but what if the error starts popping up after some time (it happened previously).

**Comment by juliasilge** [comment]
So it looks like you can run the simple code in a fresh session without observing a problem, which is good news!

Can you share a set of reproducible steps that gets you to the "FigureCanvasAgg is non-interactive, and thus cannot be shown" warning? If you can share a set of steps that causes you to see this warning, then we can dig into what may be causing it.

**Comment by manwarA** [comment]
Thank you for your time and input. The error is somehow randomly appeared, so sharing step-by-step commands/method is quiet difficult. However, I will keep an eye for the next event and will try to share all the steps. Meanwhile, I will move to new session/window to see how long will it take to re-appear the error. I hope, it will never happen again.

**Comment by juliasilge** [comment]
Let us know if you have further questions!
